### ISSUE TITLE
Enum leads to ValueError when loading OpenAPI-Spec with FastAPI

### ISSUE THREAD

**Issue by mawoka-myblock** [collerek/ormar#699]
**Describe the bug**
If a model includes an Enum, everything works as expected untill I try to open the OpenAPI-Specs generated by FastAPI.

**To Reproduce**
Create a project with ormar and FastAPI (I also use Alembic). Create the following modell:

```python
class UserAuthTypes(Enum):
    LOCAL = "LOCAL"
    GOOGLE = "GOOGLE"


class User(ormar.Model):
    id: uuid.UUID = ormar.UUID(primary_key=True, default=uuid.uuid4())
    email: str = ormar.String(unique=True, max_length=100)
    username: str = ormar.String(unique=True, max_length=100)
    password: str = ormar.String(max_length=100)
    verified: bool = ormar.Boolean(default=False)
    verify_key: str = ormar.String(unique=True, max_length=100, nullable=True)
    created_at: datetime = ormar.DateTime(default=datetime.now())
    auth_type: UserAuthTypes = ormar.Enum(enum_class=UserAuthTypes, default=UserAuthTypes.LOCAL)
    google_uid: Optional[str] = ormar.String(unique=True, max_length=25, nullable=True)

    class Meta:
        tablename = "users"
        metadata = metadata
        database = database

    class Config:
        use_enum_values = True
```

Then, execute the following SQL to tell Postgres what this "userauthtypes" is:
```sql
CREATE TYPE userauthtypes AS ENUM ('LOCAL', 'GOOGLE');
```
Open the OpenAPI-Spec from FastAPI and see that it crashes. You may need to have the User-model as an input/response-class.

**Expected behavior**
It should work.

**Screenshots**
<details>
<summary>The following error occurs </summary>

```text
Traceback (most recent call last):
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/middleware/base.py", line 43, in call_next
    message = await recv_stream.receive()
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/anyio/streams/memory.py", line 94, in receive
    return self.receive_nowait()
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/anyio/streams/memory.py", line 87, in receive_nowait
    raise EndOfStream
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 372, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/applications.py", line 269, in __call__
    await super().__call__(scope, receive, send)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/middleware/sessions.py", line 80, in __call__
    await self.app(scope, receive, send_wrapper)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/middleware/base.py", line 68, in __call__
    response = await self.dispatch_func(request, call_next)
  File "/home/mawoka/PycharmProjects/ClassQuiz/./classquiz/__init__.py", line 28, in sentry_exception
    raise e
  File "/home/mawoka/PycharmProjects/ClassQuiz/./classquiz/__init__.py", line 22, in sentry_exception
    response = await call_next(request)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/middleware/base.py", line 46, in call_next
    raise app_exc
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/middleware/base.py", line 36, in coro
    await self.app(scope, request.receive, send_stream.send)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/exceptions.py", line 93, in __call__
    raise exc
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/exceptions.py", line 82, in __call__
    await self.app(scope, receive, sender)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/routing.py", line 670, in __call__
    await route.handle(scope, receive, send)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/routing.py", line 266, in handle
    await self.app(scope, receive, send)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/starlette/routing.py", line 65, in app
    response = await func(request)
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/applications.py", line 224, in openapi
    return JSONResponse(self.openapi())
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/applications.py", line 199, in openapi
    self.openapi_schema = get_openapi(
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/openapi/utils.py", line 443, in get_openapi
    return jsonable_encoder(OpenAPI(**output), by_alias=True, exclude_none=True)  # type: ignore
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/encoders.py", line 66, in jsonable_encoder
    return jsonable_encoder(
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/encoders.py", line 101, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/encoders.py", line 101, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/encoders.py", line 101, in jsonable_encoder
    encoded_value = jsonable_encoder(
  [Previous line repeated 3 more times]
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/encoders.py", line 145, in jsonable_encoder
    return jsonable_encoder(
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/encoders.py", line 145, in jsonable_encoder
    return jsonable_encoder(
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/encoders.py", line 101, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/encoders.py", line 145, in jsonable_encoder
    return jsonable_encoder(
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/encoders.py", line 145, in jsonable_encoder
    return jsonable_encoder(
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/encoders.py", line 101, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/mawoka/.local/share/virtualenvs/ClassQuiz-BX_cFwCN/lib/python3.10/site-packages/fastapi/encoders.py", line 144, in jsonable_encoder
    raise ValueError(errors)
ValueError: [TypeError("'builtin_function_or_method' object is not iterable"), TypeError('vars() argument must have __dict__ attribute')]
```
</details>

**Versions (please complete the following information):**
 - Database backend used: PostgreSQL
 - Python version: `3.10.4`
 - `ormar` version: `0.11.1`
 - `pydantic` version: `1.9.1`
 - `fastapi` version: `0.78.0`

**Additional context**
I'm not sure if ormar, FastAPI or Pydantic is responsible for this error.

**Comment by mawoka-myblock** [comment]
Here's a working and small example:
```python
import pydantic
import ormar
from fastapi import FastAPI
from enum import Enum
import sqlalchemy
import databases
class TestEnum(Enum):
    test = "test"
    lol = "lol"
class TestModel(ormar.Model):
    enum: TestEnum = ormar.Enum(enum_class=TestEnum, default=TestEnum.test)
    lol: str = ormar.Text(primary_key=True)
    class Meta:
        metadata = sqlalchemy.MetaData()
        database = databases.Database("sqlite://./lol.db")
# class TestModel(pydantic.BaseModel):
#     enum: TestEnum
#     lol: str
app = FastAPI()
@app.get("/", response_model=TestModel)
async def index():
    return TestModel(enum=TestEnum.lol, lol="s")
```

**Comment by mawoka-myblock** [comment]
I've done some more digging and found out that the `schema()`-function works, but only the `schema_json()`-function broken is.
Below are 2 schemas, the first is from PyDantic and the other from ormar.


```json
{
   "title":"LolPydantic",
   "type":"object",
   "properties":{
      "size":{
         "default":1,
         "allOf":[
            {
               "$ref":"#/definitions/MyEnum"
            }
         ]
      }
   },
   "definitions":{
      "MyEnum":{
         "title":"MyEnum",
         "description":"An enumeration.",
         "enum":[
            1,
            2
         ]
      }
   }
}
```

Here from ormar:
```diff
{
   "title":"Example3_MZU",
   "type":"object",
   "properties":{
      "size":{
         "default":1,
+         "enum_class":"<enum MyEnum>",
         "allOf":[
            {
               "$ref":"#/definitions/MyEnum"
            }
         ]
      }
   },
   "definitions":{
      "MyEnum":{
         "title":"MyEnum",
         "description":"An enumeration.",
         "enum":[
            1,
            2
         ]
      }
   }
}
```

I think that the extra line is causing the JSON-encoder to crash. What do you think, @collerek ? I've also added a failing test in https://github.com/collerek/ormar/pull/714

**Comment by collerek** [comment]
Thanks for catching that and investigating! :)
Should be fixed in 0.11.2, please update and check.
